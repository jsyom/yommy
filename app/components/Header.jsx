import React, { Component } from 'react';
import ReactDOM from 'react-dom';

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      scrolled: false,
      menuClicked: false,
      scrollPosition: 0
    };
    this.handleScroll = this.handleScroll.bind(this);
    this.scrollToTop = this.scrollToTop.bind(this);
    this.scrollToBottom = this.scrollToBottom.bind(this);
  }

  componentDidMount() {
    window.addEventListener('scroll', this.handleScroll);
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.handleScroll);
  }

  onMenuClick() {
    console.log(this.state.menuClicked);
    this.setState({
      menuClicked: !this.state.menuClicked
    });
  }

  scrollToBottom() {
    console.log('scrolling To Bottom');
    window.scrollTo(5000, 5000);
  }
  scrollToTop() {
    console.log('scrolling To Top');
    window.scrollTo(0, 0);
  }
  handleScroll() {
    if (window.scrollY > 100) {
      this.setState({ scrolled: true });
    } else if (window.scrollY < 100) {
      this.setState({ scrolled: false });
    }
  }

  render() {
    return (
      <div className="header">
        <div className='header-item'>Home</div>
        <div className='header-item'>About</div>
        <div className='header-item'>John</div>
        <div className='header-item' onClick={() => this.scrollToBottom() } >Scroll Button to Bottom</div>
        <div className='header-item' onClick={() => this.scrollToTop() } >Scroll Button to Top</div>
        <div onClick={() => this.onMenuClick()} className='menu-item'>
          {
            this.state.menuClicked
            ? <i className="fa fa-cog" aria-hidden="true"></i>
            : <i className="fa fa-bars" aria-hidden="true"></i>
          }
        </div>
      </div>
    );
  };
}

export default Header;
