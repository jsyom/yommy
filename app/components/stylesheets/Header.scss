@import 'Media.scss';

.header {
  font-family: Indie Flower;
  display: flex;
  flex-direction: row;
  align-items: center;
  color: #E50914;
    @include respond-to(handhelds) {
      margin-left: -3vw;
    };
}

.header-item {
  text-transform: uppercase;
  padding: 1rem;
}

.menu-item {
  margin-left: auto;
  padding-right: 20px;
}
