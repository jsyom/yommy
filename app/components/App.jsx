import React, { Component } from 'react';
import Header from './Header';
import Jumbo from './Jumbo';
import About from './About';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: ''
    };
  }

  render() {
    return (
      <div className="app-container">
        <Header />
        <Jumbo />
        <About />
      </div>
    );
  }
}
