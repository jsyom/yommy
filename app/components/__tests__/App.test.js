import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../App';
import Header from '../Header';

test('App component should render as expected', () => {
	const component = shallow(<App />);
	const tree = toJson(component);
	expect(tree).toMatchSnapshot();
});

test('App component should render child components', () => {
	const component = shallow(<App />);
	const tree = toJson(component);
	expect(component.contains(<Header />)).toBe(true);
	expect(tree).toMatchSnapshot();
});

// test('fetchMovie with a correct title should return the movie', () => {});
