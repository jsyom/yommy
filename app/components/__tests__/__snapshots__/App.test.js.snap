// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render as expected 1`] = `
<div
  className="app-container"
>
  <Header />
  <SearchBar
    onSearchSubmit={[Function]}
  />
  <MovieCard
    error=""
    movie={Object {}}
  />
  <ColorsList
    movie={Object {}}
  />
</div>
`;

exports[`App component should render child components 1`] = `
<div
  className="app-container"
>
  <Header />
  <SearchBar
    onSearchSubmit={[Function]}
  />
  <MovieCard
    error=""
    movie={Object {}}
  />
  <ColorsList
    movie={Object {}}
  />
</div>
`;
